lib/__test__/KModal.spec.js

```diff
@@ -0,0 +1,134 @@
+import { shallowMount } from '@vue/test-utils';
+import KModal from '../KModal';
+
+// jest.mock('kolibri.utils.coreStrings', () => {
+//   const translations = {
+//     readReference: 'Reference',
+//   };
+//   return {
+//     $tr: jest.fn(key => {
+//       return translations[key];
+//     }),
+//   };
+// });
+
+describe('KModal component', () => {
+  it(`smoke test`, () => {
+    const wrapper = shallowMount(KModal);
+    expect(wrapper.exists()).toBe(true);
+  });
+
+  describe('props', () => {
+    it(`a title should appear`, () => {
+      const title = 'test title';
+      const wrapper = shallowMount(KModal, {
+        propsData: {
+          title: title,
+        },
+      });
+      expect(wrapper.find('.title').text()).toEqual(title);
+    });
+    it(`text should appear on the submit button`, () => {
+      const submitText = 'hello';
+      const onClick = jest.fn();
+      const wrapper = shallowMount(KModal, {
+        propsData: {
+          submitText: submitText,
+        },
+        listeners: { submit: onClick },
+      });
+      expect(wrapper.findComponent({ name: 'KButton' }).props().text).toEqual(submitText);
+    });
+
+    it(`text should appear on the close button`, () => {
+      let cancelText = 'goodbye';
+      const onClick = jest.fn();
+      const wrapper = shallowMount(KModal, {
+        propsData: {
+          cancelText: cancelText,
+        },
+        listeners: { cancel: onClick },
+      });
+      expect(wrapper.findComponent({ name: 'KButton' }).props().text).toEqual(cancelText);
```

Comment:
Never again will I import the component now that I know I can pass an object with the name property!

Reply:
YES! I learned that, I think, from reading someone else's test, and 🤯 .